### PR TITLE
Rebase develop factor uq

### LINKED ISSUE #215: Repair UQ code

### PR DESCRIPTION
**Pull request by uhetmaniuk**
Closes #215 

Updated the UQ components to the develop-refactor branch
Tested compilation and ran one simulation.

### CODE REVIEW

File: src/nimble_rve.h [hunk 1]
```diff
@@ -143,19 +143,19 @@ class RVE : public Material
...
-      double                                         time_previous,
-      double                                         time_current,
-      nimble_kokkos::DeviceFullTensorSingleEntryView deformation_gradient_n,
-      nimble_kokkos::DeviceFullTensorSingleEntryView deformation_gradient_np1,
-      nimble_kokkos::DeviceSymTensorSingleEntryView  stress_n,
-      nimble_kokkos::DeviceSymTensorSingleEntryView  stress_np1)
-  {
-  }
-#endif
+//#ifdef NIMBLE_HAVE_KOKKOS
```
Comment (rjones-sandia) [inline src/nimble_rve.h:146]: just fyi it looks like this is WIP

File: src/nimble_uq.cc [hunk 2]
```diff
@@ -0,0 +1,790 @@
...
+  for (int nuq = 0; nuq < nsamples_; nuq++) {
+    offnominal_displacements_.push_back(
+        data_->GetNodeData(data_->GetFieldId("off_nom_displacement_" + std::to_string(nuq))));
+    offnominal_velocities_.push_back(data_->GetNodeData(data_->GetFieldId("off_nom_velocity_" + std::to_string(nuq))));
+    offnominal_forces_.push_back(
+        data_->GetNodeData(data_->GetFieldId("off_nom_internal_force_" + std::to_string(nuq))));
+  }
+}
+//===========================================================================
+void
```
Comment (rjones-sandia) [inline src/nimble_uq.cc:422]: these parts of the verlet integrator can be moved into the uq block now that it see the off_nom data

**Review by rjones-sandia — APPROVED** [review]
nice replacement of the ifdef shoehorn coding
- it looks like the current code now largely relies on whether the "off_nom" disp/vel etc have been allocated
- the generality of this function ParseMaterialParamsStringToMap is welcome
- the design of UqModel should change to reflect how uq_model_data/uq_block act e.g. moving the verlet integration of the velocites and the like but this doesn't need to happen now

**Comment by uhetmaniuk** [comment]
@rjones-sandia 
Thank you for reviewing the PR.
If that's ok with you, I would prefer to first merge this PR and then work on the "rebase" of develop-refactor.
I will create a separate issue with your bullets as a next task.
Please let me know if you have a concern with this plan.

**Comment by rjones-sandia** [comment]
Hi Ulrich
  yes when I was getting dinner I realized my comments were ambiguous.  Please proceed with the rebase of the develop-refactor into develop. I'll build the UQ enabled code in the meantime.
Reese

**Pull request merged**